hudi-common/src/main/java/org/apache/hudi/exception/SchemaUnbalancedUnionException.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.hudi.exception;
+
+/**
+ * Thrown when a field is a union and at least one of the following is true:
+ * <ul>
+ *   <li>the incoming union and the latest table union have differing numbers of types</li>
+ *   <li>the incoming union has more than two types</li>
+ * </ul>
+ */
+public class SchemaUnbalancedUnionException extends SchemaCompatibilityException {
```

Comment:
rename to `InvalidUnionTypeException`?

Reply:
Still not sure what `Unbalanced` means in the naming.  Use `Invalid` is better?  Or do we need this specific exception (again going back to the point if you can directly throw `SchemaCompatibilityException` with a specific message)?